src/domain/staking/staking.repository.ts

```diff
@@ -93,6 +102,40 @@ export class StakingRepository implements IStakingRepository {
     return DefiVaultsStateSchema.parse(defiStats)[0];
   }
 
+  public async getDefiVaultStake(args: {
+    chainId: string;
+    safeAddress: `0x${string}`;
+    vault: `0x${string}`;
+  }): Promise<DefiVaultStake> {
+    const stakingApi = await this.stakingApiFactory.getApi(args.chainId);
+    const defiStakes = await stakingApi.getDefiVaultStakes(args);
+    // Cannot be >1 contract deployed at the same address so return first element
```

Comment:
Not sure I understand this comment. We are fetching the deposits on a specific vault (1 address) for a specific safe so the response array always contains exactly one element right?